### PR TITLE
Script processing refactoring and optimization, part2

### LINKED ISSUE #15: Figure out why we need JSON processing. And remove it if it's unnecessary.
I've also seen a part of it in the `js-processor`

### PR DESCRIPTION
**Pull request by inikulin**
- remove JSON processing (fixes #15)
- Join script processor and js processor and make it just set of
  processing utility methods in src/processing/script
- Better handling of the array object script, tests added as well
- Processing header tests
- Remove jsProcessor from client API, expose hammerhead.navigateTo(url)
  instead

\cc @churkin @LavrovArtem 
@VasilyStrelyaev please check test names and comments

### CODE REVIEW

File: src/processing/script/header.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    `${ INSTRUCTION.setProperty }=__w$?__w$.${ INSTRUCTION.setProperty }:function(o,p,v){return o[p] = v},`,
+    `${ INSTRUCTION.getProperty }=__w$?__w$.${ INSTRUCTION.getProperty }:function(o,p){return o[p]},`,
+    `${ INSTRUCTION.callMethod }=__w$?__w$.${ INSTRUCTION.callMethod }:function(o,p,a){return o[p].apply(o,a)},`,
+    `${ INSTRUCTION.processScript }=__w$?__w$.${ INSTRUCTION.processScript }:function(s){return s};`,
+    POSTFIX,
+    '\n'
+].join('');
+
+// NOTE: IE remove trailing new lines in `script.textContent`,
+// so trailing new line in RegExp is optional
```
Comment (VasilyStrelyaev) [inline src/processing/script/header.js]: IE removes trailing newlines in `script.textContent`, so a trailing newline in RegExp is optional

File: src/client/index.js [hunk 2]
```diff
@@ -118,6 +119,11 @@ class Hammerhead {
             eventOwner.off(evtName, handler);
     }
 
+    navigateTo (url) {
+        window.location = getProxyUrl(url);
```
Comment (churkin) [inline src/client/index.js]: It is more logical to use the window that is passed in the start method

**Comment by churkin** [comment]
lgtm

**Comment by inikulin** [comment]
FPR

File: src/processing/script/index.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+
+// Parse/generate code
+function removeTrailingSemicolonIfNecessary (processed, src) {
+    return TRAILING_SEMICOLON_RE.test(src) ? processed : processed.replace(TRAILING_SEMICOLON_RE, '');
+}
+
+function getAst (src, isObject) {
+    // NOTE: In object case (e.g. `eval('{ 1: 2}')` ) without wrapping
```
Comment (VasilyStrelyaev) [inline src/processing/script/index.js]: `// NOTE: In case of objects (e.g. `eval('{ 1: 2}')` ) without wrapping`

**Comment by inikulin** [comment]
Client tests just pissing me off today. Gosh...

**Comment by inikulin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**